raft-tech/TANF-app
# 2509 - Data submission email

## Summary of Changes
Pull request closes #2509. Updates the `send_data_submitted_email` to
* include more information in the template
* send once parsing completes, rather when file upload completes
* distinguish "processed with errors" and "processed without errors" in the subject and preview line

## How to Test

1. Update `SENDGRID_API_KEY` in your `.env` to a valid api key (i can provide one if you reach out). ensure `SENDGRID_SANDBOX_MODE_IN_DEBUG` remains `False`.
1. Run
   ```
   cd tdrs-backend && docker-compose up
   cd tdrs-frontend && docker-compose up --build
   ```
1. Set your account type to "Data Analyst"
1. Upload a data file (both containing errors and containing no errors)
1. Inspect the emails you receive

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Aligned with design spec ("Near Term Lower-lift" version):
   - [x] Data analysts receive email notification that their data is processed
   - [x] Email subject leads with whether or not their data file had errors
   - [x] Email table contains columns for section, submitted by, and error report (Error Report Available/No Errors being possible values)
   - [x] Email body contains copy on where to access the error report in TDP and where to go for more information in the knowledge center
- [x] Replaces existing notifications for data file submission

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
